extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/asset/AssetUpdater.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2022 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.pages.asset;
+
+import de.sovity.edc.ext.wrapper.api.common.mappers.AssetMapper;
+import de.sovity.edc.ext.wrapper.api.common.mappers.utils.EdcPropertyUtils;
+import de.sovity.edc.ext.wrapper.api.common.model.UiAssetCreateRequest;
+import de.sovity.edc.ext.wrapper.api.common.model.UiAssetEditMetadataRequest;
+import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.SelfDescriptionService;
+import lombok.RequiredArgsConstructor;
+import org.eclipse.edc.spi.types.domain.asset.Asset;
+
+@RequiredArgsConstructor
+public class AssetUpdater {
+    private final AssetMapper assetMapper;
+    private final EdcPropertyUtils edcPropertyUtils;
+    private final AssetIdValidator assetIdValidator;
+    private final SelfDescriptionService selfDescriptionService;
+
+    /**
+     * Creates an {@link Asset} from a {@link UiAssetCreateRequest}.
+     *
+     * @param request {@link UiAssetCreateRequest}
+     * @return {@link Asset}
+     */
+    public Asset createAsset(UiAssetCreateRequest request) {
```

Comment:
The class is called `AssetUpdater` but contains a method that doesnt have an asset argument and instead creates one from `UiAssetCreateRequest`